src/backend/marsha/development/views.py

```diff
@@ -48,15 +49,20 @@ def get_context_data(self, **kwargs):
 
         """
         domain = self.request.build_absolute_uri("/").split("/")[2]
+
+        # use the HTTP_REFERER like to be consistent with the LTI passport
+        request_domain = urlparse(self.request.build_absolute_uri("/")).hostname
         try:
-            consumer_site = ConsumerSite.objects.get(domain=domain)
+            consumer_site = ConsumerSite.objects.get(domain=request_domain)
         except ConsumerSite.DoesNotExist:
             consumer_site, _ = ConsumerSite.objects.get_or_create(
-                domain=domain, name=domain
+                domain=request_domain, name=request_domain
             )
 
         try:
-            playlist = Playlist.objects.get(consumer_site=consumer_site)
+            playlist = Playlist.objects.filter(consumer_site=consumer_site).latest(
+                "created_on",
+            )
```

Comment:
No exception will be thrown here. You should test if `playlist` is `None` instead and remove the try/except block

```
if not (playlist := Playlist.objects.filter(consumer_site=consumer_site).latest("created_on")):
    # create it
```

WDYT ?

Reply:
Well spotted :+1:

Reply:
Actually, `latest` would raise, I replaced it with your code, using `last()` instead.